cmd/internal/loginprovider/tryinorderloginprovider.go

```diff
@@ -21,14 +21,27 @@ import (
 // for the first time when we still don't know which login method.
 func NewTryInOrderLoginProvider(logger loggo.Logger, providers ...api.LoginProvider) api.LoginProvider {
 	return &tryInOrderLoginProviders{
-		providers: providers,
-		logger:    logger,
+		providers:  providers,
+		logger:     logger,
+		loginToken: missingToken,
 	}
 }
 
+func missingToken() (string, error) {
+	return "", api.ErrorLoginFirst
+}
```

Comment:
Here we set the default return result for the InOrderLoginProvider to the `ErrorLoginFirst` error. If the login is successful we replace this with a closure to the successful providers Token method.